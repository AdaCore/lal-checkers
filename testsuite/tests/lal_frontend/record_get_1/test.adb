procedure Ex1 is
   type Point is record
      x : Integer;
      y : Integer;
   end record;
   type Rectangle is record
      top_left: Point;
      bottom_right: Point;
   end record;
   r : Rectangle;
   z : Boolean := False;
begin
   if (r.top_left.x = 2 and
       r.top_left.y = 3 and
       r.bottom_right.x = 12 and
       r.bottom_right.y = 8) then
      z := True;
   end if;
end Ex1;
