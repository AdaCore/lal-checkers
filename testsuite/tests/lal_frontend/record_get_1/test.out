Program:
  r#1 = Input()
  z#2 = False
  split:
    assume(&&(&&(&&(==(Get_0(Get_0(r#1)), 2), ==(Get_1(Get_0(r#1)), 3)), ==(Get_0(Get_1(r#1)), 12)), ==(Get_1(Get_1(r#1)), 8)))
    z#2 = True
  |:
    assume(!(&&(&&(&&(==(Get_0(Get_0(r#1)), 2), ==(Get_1(Get_0(r#1)), 3)), ==(Get_0(Get_1(r#1)), 12)), ==(Get_1(Get_1(r#1)), 8))))
  end:
